### DISCUSSION TITLE
Rules not showing

### DISCUSSION THREAD

**Discussion by DengelFred** [O-X-L/ansible-opnsense#76] [Q&A] [answered]
Hello,

just tried this role, but it returns nothing. Even not with 'syslog'... any hint whats going wrong?
Of course, I  have a few rules defined ;-)

Thanks a lot :-)

```
- name: Test OpnSense API Call
  hosts: opnsense
  gather_facts: true

  module_defaults:
    group/ansibleguy.opnsense.all:
      api_key: "{{ opnsense_api_key }}"
      api_secret: "{{ opnsense_api_secret }}"
      api_retries: 10
      firewall: "{{ opnsense_fqdn }}"
      ssl_verify: false  # TODO: Add valid certificates to the basic config

  tasks:
    - name: get current rules
      ansibleguy.opnsense.list:
        target: 'rule'
      register: existing_entries

    - debug: var=existing_entries
```

This is the result:
```
PLAY [Test OpnSense API Call] ************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************
[WARNING]: Platform freebsd on host opnsense is using the discovered Python interpreter at /usr/local/bin/python3.11, but
future installation of another Python interpreter could change the meaning of that path. See
https://docs.ansible.com/ansible-core/2.16/reference_appendices/interpreter_discovery.html for more information.
ok: [opnsense]

TASK [get current rules] *****************************************************************************************************
ok: [opnsense]

TASK [debug] *****************************************************************************************************************
ok: [opnsense] => {
    "existing_entries": {
        "changed": false,
        "data": [],
        "failed": false
    }
}

PLAY RECAP *******************************************************************************************************************
opnsense                   : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

**Comment by ansibleguy** [comment]
Greetings.

I think your problem is not using `connection: local` on the playbook: https://opnsense.ansibleguy.net/en/latest/usage/2_basic.html#inventory-playbook

As we are communicating with an API - the module should be executed on the Ansible controller:
![image](https://github.com/ansibleguy/collection_opnsense/assets/79044229/d895bcd9-5e1e-41c4-96f6-45c2a325fafa)

**Comment by DengelFred** [comment]
Oh yes, you are right. Makes no sense to run a API access from remote host ;-)
However, problem is still the same. Is it possible, that the "rule" request only returns the rules from the automate view?
But how to reach all the other, manual rules? This must be handled directly in the config.xml, right?

**Reply by superstes** [reply, marked as answer]
Yes. They are separated for some reason. This is by design and we can't manage the other rulsets as they are not API enabled.

https://opnsense.ansibleguy.net/en/latest/modules/rule.html#limitations

`the ruleset managed by this plugin is SEPARATE from the default WEB-UI rules (Firewall - Rules) - combined usage might bring complications`
